lib/liquid/filters.ex

```diff
@@ -27,34 +31,436 @@ defmodule Liquid.Filters do
+    def modulo(input, operand) do
+      input |> to_number |> modulo(to_number(operand))
+    end
+
+
+    def truncate(input, l \\ 50)
+
+    def truncate(nil, _), do: nil
+
+    def truncate(input, l) do
+      truncate_string = "..."
+      l = l - String.length(truncate_string) - 1
+      case {l, String.length(input)} do
+        {l, _} when l <= 0 -> truncate_string
+        {l, len} when l < len -> String.slice(input,0..l) <> truncate_string
+        _ -> input
+      end
+    end
+    
+    def truncatewords(input, words \\ 15)
+
+    def truncatewords(nil, _), do: nil
+
+    def truncatewords(input, words) do
+      truncate_string = "..."
+      wordlist = input |> String.split(" ")
+      case words - 1 do
+        l when l < 0 -> wordlist[0]
+        l when l < length(wordlist) ->
+          words = wordlist |> Enum.slice(0..l) |> Enum.join(" ") 
+          words <> truncate_string
+        _ -> input
+      end
+    end
+
 
     def replace(<<string::binary>>, <<from::binary>>, <<to::binary>>) do
       string |> String.replace(from, to)
     end
 
+    def replace(<<string::binary>>, <<from::binary>>, to) do
+      string |> replace(from, to_string(to))
+    end
+
+    def replace(<<string::binary>>, from, to) do
+      string |> replace(to_string(from), to)
+    end
+
+    def replace(string, from, to) do
+      string |> to_string |> replace(from, to)
+    end
+
+    def replace_first(<<string::binary>>, <<from::binary>>, to) do
+      string |> String.replace(from, to_string(to), global: false)
+    end
+
+    def replace_first(string, from, to) do
+      to = to |> to_string
+      string |> to_string |> String.replace(to_string(from), to, global: false)
+    end
+
+    def remove(<<string::binary>>, <<remove::binary>>) do
+      string |> String.replace(remove, "")
+    end
+
+    def remove_first(<<string::binary>>, <<remove::binary>>) do
+      string |> String.replace(remove, "", global: false)
+    end
+
+    def remove_first(string, operand) do
+      string |> to_string |> remove_first(to_string(operand))
+    end
+
+    def append(<<string::binary>>, <<operand::binary>>) do
+      string <> operand
+    end
+
+    def append(input, nil), do: input
+    
+    def append(string, operand) do
+      string |> to_string |> append(to_string(operand))
+    end
+
+
     def prepend(<<string::binary>>, <<addition::binary>>) do
       addition <> string
     end
 
-    def prepend(<<string::binary>>, nil), do: string
+    def prepend(string, nil), do: string
+
+    def prepend(string, addition) do
+      string |> to_string |> append(to_string(addition))
+    end
+
+    def strip(<<string::binary>>) do
+      string |> String.trim
+    end
+
+    def lstrip(<<string::binary>>) do
+      string |> String.trim_leading
+    end
+
+    def rstrip(<<string::binary>>) do
+      string |> String.trim_trailing
+    end
+
+    def strip_newlines(<<string::binary>>) do
+      string |> String.replace("\n", "") |> String.replace("\r", "")
```

Comment:
I wonder if this will give problems when someone stores data with windows file ending \r\n